File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int count = nums.length/2; // 뽑을 수 있는 횟수
        Set<Integer> reNums = new HashSet();
        
        for (int i = 0; i < nums.length; i++) {
            reNums.add(nums[i]);
        }
        
        return (int) Math.min(count, reNums.size());
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    var count = nums.length/2; // 가져갈 수 있는 마리수
    var reNums = new Set(nums).size; // 포켓몬 종류 중복제거
    
    // 3항 연산자를 이용하여 더 작은 수를 반환
    return count > reNums ? reNums : count
}